FoodplannerApi/Controller/UsersController.cs

```
@@ -225,4 +225,25 @@ public async Task<IActionResult> UpdatePassword([FromHeader(Name = "Authorizatio
         return NotFound();
     }
 
+    [HttpDelete]
+    [Authorize(Roles = "Parent, Child, Teacher, Admin")]
+    [ProducesResponseType(StatusCodes.Status204NoContent)]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    public async Task<IActionResult> DeleteLoggedInUser([FromHeader(Name = "Authorization")] string token)
+    {
+        var idString = _authService.RetrieveIdFromJwtToken(token);
+        if (!int.TryParse(idString, out int id))
+        {
+            return BadRequest(new ErrorResponse { Message = ["Id er ikke et tal"] });
+        }
+
+        int result = await _userService.DeleteUserAsync(id);
+
+        if (result > 0)
+        {
+            return NoContent();
```

I think this should either return a exception or a error response.

"Id couldn't be found"

---

Der bliver allerede sendt en error response hvis Id ikke parser og en 404 status hvis det fejler.